modules/data.remote/inst/SMAPvsModel_Comparison.R

```diff
@@ -0,0 +1,353 @@
+##'5. File location for output of NEON site geoJson files
+##'6. File location for output of SMAP netCDF files
+##'
+##'@author Joshua Bowers
+##'
+##'@examples
+##'
+##'# Run all CONUS NEON sites at once
+##'all_sites <- PEcAn.settings::read.settings("~/pecan.xml")
+##'sites_vector <- c()
+##'for (index in 1:length(all_sites)) {
+##'sites_vector <- append(sites_vector, all_sites[[index]]$run$site$id) 
+##'}
+##'sites_vector <- sites_vector[c(-4, -33, -37)] # Removes D17 sites that are not collected by SMAP
+##'main(sites_vector, compare = TRUE, pdf_output = '/projectnb/dietzelab/jbowers1/graphic_output/')\
+##'
+##'# Or run individual outputs printed to your screen
+##'var <- runAnalysis('1000004945') # harvard
+##'var1 <- runAnalysis('1000004876') # santa rita
+##'var2 <- runAnalysis('1000004927') # konza
+##'var3 <- runAnalysis('1000004916') # ordway
+##'
+##'
+
+# load necessary packages
+library(tidyverse)
+library(plotly)
+library(patchwork)
+library(babynames)
+library(viridis)
+library(purrr)
+library(lubridate)
+library(tidyr)
+library(dplyr)
+library(ncdf4)
+library(stringr)
+library(hydroGOF)
+library(Metrics)
+library(verification)
+
+
+#################### INPUTS #################### 
+## Set Date Range ##
+start <- '2016-01-02'
+end <- '2016-07-16'
+
+## Directory of Ensembles Files ##
+ensamble_dir <- '/projectnb2/dietzelab/dongchen/All_NEON_SDA/NEON42/SDA_testrun/out/'
+
+## CONUS NEON Sites XML File Location ##
+input_dir <- '~/pecan/modules/data.remote/inst/conus_sites.xml'
+
+## File Directory to Place Output PDFs In ##
+output_dir <- '/projectnb/dietzelab/jbowers1/graphic_output/'
+
+## Output Directories for geoJson Files ##
+geoJSON_outdir <- '/projectnb/dietzelab/jbowers1/geoFiles/'
+
+## Output Directories for SMAP netCDF Files ##
+smap_outdir <- '/projectnb/dietzelab/jbowers1/smap_ncFiles/'
+
+## Download SMAP data ##
+source('~/pecan/modules/data.remote/inst/download_SMAP.R')
+all_smap_data <- download_SMAP(start, end, geoJSON_outdir, smap_outdir)
+####################^ INPUTS ^####################
+start <- as.Date(start)
+end <- as.Date(end)
+
+
+#' @title extractSoilMoist
+#' @description extracts a single ensamble member during the given year range from a specified file location
+#'
+#' @param betySiteId bety site id for the desired location (String)
+#' @param yearStart starting year YYYY as (date) or (numeric)
+#' @param yearEnd ending year YYYY as (date) or (numeric)
+#' @param ENS ensamble member front padded with 0's to five values (String)
+#'
+#' @return A list containing soil moisture and time values
+#'
+#' @examples 
+#' extractSoilMoist(betySiteId = '1000004945', yearStart = 2016, yearEnd = 2017, ENS = '25')
+#' 
+#' This extracts soil moisture values and time values from the 25th ensamble member netCDF file 
+#' in a specified file directory for the Harvard Forest ('1000004945') site during the years 
+#' 2016 and 2017.
+#' 
+extractSoilMoist <- function(betySiteId, yearStart, yearEnd, ENS){
```

Comment:
There's an argument to be made that a lot of these functions should be moved to the R directory so that they are more generally useable. If so, an important question is whether a function like this, which operates on model output not data, would be better placed in a different module

Reply:
New module called netCDFvarExtraction created with the two functions extractPredictionVar (previously extractEnsemble) and extractVariableENS (previously extractSoilMoist).  extractPredictionVar and extractVariableENS have a new arg called predictVar that allows the user to enter the variable to extract (ex. "SoilMoist" for soil moisture values).

Reply:
@jbowers20 while you pulled a few functions out of your main validation script, you kept them in the /inst directory so they're still not an exported part of the package.